### PR TITLE
Make computeAlignment more efficient

### PR DESCRIPTION
**Pull request by Bodigrim**
Currently `Test.Tasty.Ingredients.ConsoleReporter.computeAlignment` evaluation is extremely inefficient: firstly because `data Maximum` lacks strictness annotations, and secondly because `foldTestTree` operates not even over `Maximum a`, but over `Int -> Maximum a`.

The patch improves efficiency of `computeAlignment` and arguably makes its implementation simpler.

CC @martijnbastiaan for review.

### CODE REVIEW

File: core/Test/Tasty/Ingredients/ConsoleReporter.hs [hunk 1]
```diff
@@ -749,35 +749,19 @@ formatDesc n desc =
...
-      { foldSingle = \_ name _ level -> Maximum (stringWidth name + level)
-      , foldGroup = \_opts _ m -> mconcat m . (+ indentSize)
+  max 0 .
+  foldTestTree0
+    minBound
+    TreeFold
+      { foldSingle = \_ name _ -> stringWidth name
+      , foldGroup = \_ _ m -> if null m then minBound else maximum m + indentSize
+      , foldResource = \_ _ f -> f $ throwIO NotRunningTests
+      , foldAfter = \_ _ _ b -> b
```
Comment (VictorCMiraldo) [inline core/Test/Tasty/Ingredients/ConsoleReporter.hs:762]: I think we could make this patch even smaller borrowing the `trivialFold` from the original function:

```suggestion
    trivialFold
      { foldSingle = \_ name _ -> stringWidth name
      , foldGroup = \_ _ m -> if null m then minBound else maximum m + indentSize
```

**Review by VictorCMiraldo — APPROVED** [review]
LGTM! Thanks for the patch @Bodigrim

File: core/Test/Tasty/Ingredients/ConsoleReporter.hs (hunk 1 above)
Comment (Bodigrim) [reply core/Test/Tasty/Ingredients/ConsoleReporter.hs:762]: `trivialFold` unfortunately requires `Monoid` constraint, which we don't have here.

Comment (VictorCMiraldo) [reply core/Test/Tasty/Ingredients/ConsoleReporter.hs:762]: Too bad; ignore my suggestion then :)

**Review by martijnbastiaan — APPROVED** [review]
Looks much simpler!

**Pull request merged**